### PR TITLE
docs: add examples on how to test libraries

### PR DESCRIPTION
**Pull request by codyoss**
This is the based off the doc added to the gapic libraries. Because
of the design of the api and the fact these clients use HTTP and
not gRPC it is useful to see how testing is done here as well.

Related: googleapis/google-cloud-go#2921

### CODE REVIEW

File: internal/examples/fake/fake.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+)
+
+// TranslateText translates text to the given language using the provided
+// service.
+func TranslateText(service *translate.Service, text, language string) (string, error) {
+	parent := fmt.Sprintf("projects/%s/locations/global", os.Getenv("GOOGLE_CLOUD_PROJECT"))
+	resp, err := service.Projects.Locations.TranslateText(parent, &translate.TranslateTextRequest{
+		TargetLanguageCode: language,
+		Contents:           []string{text},
+	}).Do()
```
Comment (tbpg) [inline internal/examples/fake/fake.go:23]: Consider a separate `req` variable. Sometimes, it can be hard to follow multi-line function calls + declarations.

File: testing.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
+# Testing Code that depends on Go Client Libraries
```
Comment (tbpg) [inline testing.md:1]:
```suggestion
# Testing Code that depends on google.golang.org/api
```

**Review by tbpg — APPROVED** [review]

File: internal/examples/fake/fake.go (hunk 1 above)
Comment (codyoss) [reply internal/examples/fake/fake.go:23]: Done

File: testing.md (hunk 2 above)
Comment (codyoss) [reply testing.md:1]: Done
